src/coreComponents/physicsSolvers/simplePDE/temporaryUtilities/SubRegionMeshUtilities.hpp

```diff
@@ -0,0 +1,306 @@
+  // constructor
+  isoparametricPyramidMesh( arrayView2d< real64 const, nodes::REFERENCE_POSITION_USD > const X,
+                            ARRAY_VIEW_TYPE const elementToNodes ): isoparametricMesh< ARRAY_VIEW_TYPE >( X, elementToNodes )
+  {}
+
+  //
+  PyramidCell getCell( CellIndexType k ) const
+  {
+    real64 xLocal[numCellVertex][3]{};
+
+    for( int i=0; i<numCellVertex; ++i )
+    {
+      localIndex const localNodeIndex = this->m_elementToNodes( k, i );
+      xLocal[ i ][ 0 ] = this->m_X[ localNodeIndex ][ 0 ];
+      xLocal[ i ][ 1 ] = this->m_X[ localNodeIndex ][ 1 ];
+      xLocal[ i ][ 2 ] = this->m_X[ localNodeIndex ][ 2 ];
+    }
+
+    return PyramidCell( xLocal );
+  }
+
+  // localInde{x numCells() const;
+};
+
+
+// // Factory selectign dynamically the appropriate isopametric mesh
+// template< typename ARRAY_VIEW_TYPE >
+// isoparametricMesh< ARRAY_VIEW_TYPE > selectIsoparametricMesh( ElementType elemType,
+//                                                               arrayView2d< real64 const, nodes::REFERENCE_POSITION_USD > const
+// nodePositions,
+//                                                               ARRAY_VIEW_TYPE const elementToNodes )
+// {
+//   if( elemType == ElementType::Hexahedron )
+//   {
+//     return isoparametricHexahedronMesh< ARRAY_VIEW_TYPE >( nodePositions, elementToNodes );
+//   }
+//   // else if( elemType == ElementType::Wedge) )
+//   // {
+//   // return isoparametricWedgenMesh( nodePositions, elementToNodes );
+//   // }
+//   else
+//   {
+//     GEOS_ERROR( "finiteElement::dispatchlowOrder3D() is not implemented for input of " << elemType );
+//     return isoparametricHexahedronMesh< ARRAY_VIEW_TYPE >( nodePositions, elementToNodes );
+//   }
+// }
+
+// ***************************************************
+
+template< typename ARRAY_VIEW_TYPE, int NUM_VERTEX >
+struct NumVertexToSubregionMesh
+{
+  using type = isoparametricCuboidMesh< ARRAY_VIEW_TYPE >;
+};
+
+// template< typename ARRAY_VIEW_TYPE >
+// struct NumVertexToSubregionMesh< ARRAY_VIEW_TYPE, 6 >
+// {
+//   using type = isoparametricWedgeMesh< ARRAY_VIEW_TYPE >;
+
+template< typename ARRAY_VIEW_TYPE >
+struct NumVertexToSubregionMesh< ARRAY_VIEW_TYPE, 4 >
+{
+  using type = isoparametricTetrahedronMesh< ARRAY_VIEW_TYPE >;
+};
+
+template< typename ARRAY_VIEW_TYPE >
+struct NumVertexToSubregionMesh< ARRAY_VIEW_TYPE, 5 >
+{
+  using type = isoparametricPyramidMesh< ARRAY_VIEW_TYPE >;
+};
+
+template< typename ARRAY_VIEW_TYPE >
+struct NumVertexToSubregionMesh< ARRAY_VIEW_TYPE, 6 >
+{
+  using type = isoparametricWedgeMesh< ARRAY_VIEW_TYPE >;
+};
+// };
+
+// #define USE_IJK_MESH
+
+#ifdef USE_IJK_MESH
+template< typename ARRAY_VIEW_TYPE >
+struct NumVertexToSubregionMesh< ARRAY_VIEW_TYPE, 8 >
+{
+  using type = IJKMesh< ARRAY_VIEW_TYPE >;
+};
+#else
+template< typename ARRAY_VIEW_TYPE >
+struct NumVertexToSubregionMesh< ARRAY_VIEW_TYPE, 8 >
+{
+  using type = isoparametricCuboidMesh< ARRAY_VIEW_TYPE >;
+};
+#endif
```

Comment:
The use of a structured mesh has being tested by defining the macro `USE_IJK_MESH`. The `IJKMesh` iterable returns a cube with constant size equal to 0.1:

https://github.com/GEOS-DEV/GEOS/blob/a85ab6f7b6b93ea9a9842ccfacec1085298dfa6f/src/coreComponents/physicsSolvers/simplePDE/temporaryUtilities/SubRegionMeshUtilities.hpp#L59-L66